src/Symfony/Component/Translation/Tests/Extractor/Visitor/AbstractVisitorTest.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace Symfony\Component\Translation\Tests\Extractor\Visitor;
+
+use PhpParser\NodeVisitor;
+use Symfony\Bundle\FrameworkBundle\Tests\TestCase;
+use Symfony\Component\Translation\Extractor\PhpAstExtractor;
+use Symfony\Component\Translation\MessageCatalogue;
+
+abstract class AbstractVisitorTest extends TestCase
+{
+    abstract public function getVisitors(): NodeVisitor;
+    abstract public function getResource(): iterable|string;
+    abstract public function assertCatalogue(MessageCatalogue $catalogue): void;
+
+    public function testVisitor()
+    {
+        $extractor = new PhpAstExtractor([$this->getVisitors()]);
```

Comment:
```suggestion
        $extractor = new PhpAstExtractor($this->getVisitors());
```

I would make `getVisitors` returning an array, no?

Reply:
Or name it `getVisitor`, depending on cases where we need multiple visitors.

Reply:
let's use `getVisitor`. I renamed it when the `TranslatableMessageVisitorTest` needed both `TranslatableMessageVisitor` and `TransMethodVisitor` to work, but forgot to rename it after fixing it.